tests/parsec-cli-tests.sh

```diff
@@ -81,13 +81,45 @@ test_crypto_provider() {
     fi
 
     test_encryption
+    test_decryption
     test_signing "RSA"
     test_signing "ECC"
     test_csr "RSA"
     test_csr "ECC"
 }
 
 test_encryption() {
+    KEY="anta-key-rsa-encrypt"
+    TEST_STR="$(date) Parsec public key encryption test"
+
+    create_key "RSA" $KEY "CRYPT"
+
+    # If the key was successfully created and exported
+    if [ -s ${MY_TMP}/${KEY}.pem ]; then
+        debug cat ${MY_TMP}/${KEY}.pem
+
+        echo
+        echo "- Encrypting \"$TEST_STR\" string using Parsec public key encryption"
+
+        # Encrypt TEST_STR with the public key using Parsec rather than openssl
+        # (No need to base64 encode this, because parsec-tool already does it)
+        run_cmd $PARSEC_TOOL_CMD encrypt --key-name $KEY "$TEST_STR" > ${MY_TMP}/${KEY}.enc
+
+        echo
+        echo "- Using Parsec to decrypt the result (with the private key):"
+        run_cmd $PARSEC_TOOL_CMD decrypt $(cat ${MY_TMP}/${KEY}.enc) --key-name $KEY \
```

Comment:
We used openssl for the decryption test not only because parsec-tool didn't support encryption, but also as a proof that results of Parsec operations are compatible with third party tools and vice versa.
Maybe we should change or expand this test to include:
- generate a key pair with openssl
- import the public key into Parsec
- encrypt with Parsec using the imported key
- decrypt the result with openssl

Reply:
By "non-critical" I simply mean that it is working with the public key and therefore, even if it behaved incorrectly, one could easily work around that. It's only a convenience function. Testing it like this is better than not testing it at all.

Reply:
There'a already an open issue to add key import support: https://github.com/parallaxsecond/parsec-tool/issues/75 - I will add some commentary there to enhance testing to cover cross-testing of parsec-tool with openssl for these cases.